### DISCUSSION TITLE
dynamic route not getting rendered

### DISCUSSION THREAD

**Discussion by dagda1** [remix-run/remix#5778] [Q&A]
[Here](https://stackblitz.com/edit/node-phuvuy) is a stackblitz and I have this set up of routes:

![slug](https://user-images.githubusercontent.com/118328/224678522-72bef5d2-4109-46fc-bb3c-55b13715d482.png)

I would expect a url of `/posts/something` to render the `$slug` file but instead a **404 Not Found** is returned.

This is my first time using remix but I'm not sure what I am doing wrong.

**Comment by dcmusiclive-team** [comment]
Not an answer, but I'm running into this same issue after following the Blog Tutorial and getting to the [Dynamic Route Params](https://remix.run/docs/en/1.14.3/tutorials/blog#dynamic-route-params) part. 

I've also been able to reproduce this with the following steps:
1.  Run `npx create-remix@latest --template remix-run/indie-stack blog-tutorial` and follow prompts
2. Create a `posts` folder with a file called `$slug.tsx`
3. Paste the code snippet to app/routes/posts/$slug.tsx:
```
export default function PostSlug() {
  return (
    <main className="mx-auto max-w-4xl">
      <h1 className="my-6 border-b-2 text-center text-3xl">
        Some Post
      </h1>
    </main>
  );
}
```
4. Try to go to localhost:3000/routes/some-slug-here --> Redirected to 404 with the console saying `No routes matched location`

**Reply by kiliman** [reply]
FYI: `routes` is not part of the URL. That's simply the folder where all your routes are stored. You should try to access `/posts/some-slug-here`

**Reply by dcmusiclive-team** [reply]
Mistyped. `/posts/some-slug-here` was the route I accessed.

**Comment by kiliman** [comment]
The reason why it's not working is that you've enabled `v2_routeConvention`, but you're not using it properly. Once you use folders, *every* route must be a folder, and the route module is the file named `route.tsx`.

I've updated your stackblitz
https://stackblitz.com/edit/node-o2w958?file=app%2Froutes%2Fposts%2F%24slug%2Froute.tsx

![image](https://user-images.githubusercontent.com/47168/228549167-aa01e0ad-6689-4fc8-a4fe-bf178a9289e8.png)

Your other option is to use flat files (without any folders). These two are equivalent.

![image](https://user-images.githubusercontent.com/47168/228550067-a58134ad-da38-4b18-bf36-2210beefa663.png)

**Reply by dagda1** [reply]
@kiliman this was indeed my problem so the docs I was looking at were never going to work.

I found [this link](https://remix.run/docs/en/main/file-conventions/route-files-v2) which showed me how to do this

![routes](https://user-images.githubusercontent.com/118328/228553061-2f88dd79-215b-4238-bce0-7a1741e50ee8.png)

**Reply by Putnug1122** [reply]
in remix 1.15 folder one doesn't work anymore
![image](https://user-images.githubusercontent.com/62296958/232349806-b45f4fe7-64b4-4bac-b30f-d4828c36cd29.png)
![image](https://user-images.githubusercontent.com/62296958/232349835-81db8e9c-920a-4b50-849e-7b628b6cd317.png)

but the file works

**Reply by kiliman** [reply]
What does `npx remix routes` show?

**Reply by nlindroos** [reply]
With 
<img width="276" alt="image" src="https://user-images.githubusercontent.com/6067491/235431449-92523683-5f9a-4ae1-9038-760e1ce3a4db.png">


```zsh
npx remix routes

<Routes>
  <Route file="root.tsx">
    <Route path="healthcheck" file="routes/healthcheck.tsx" />
    <Route index file="routes/_index.tsx" />
    <Route path="logout" file="routes/logout.tsx" />
    <Route path="login" file="routes/login.tsx" />
    <Route path="notes" file="routes/notes.tsx">
      <Route path=":noteId" file="routes/notes.$noteId.tsx" />
      <Route index file="routes/notes._index.tsx" />
      <Route path="new" file="routes/notes.new.tsx" />
    </Route>
    <Route path="join" file="routes/join.tsx" />
  </Route>
</Routes>
```

**Reply by nlindroos** [reply]
I probably misintrepreted [the docs](https://remix.run/docs/en/main/file-conventions/route-files-v2), but I tried to use [the recommended folder organisation approach](https://remix.run/docs/en/main/file-conventions/route-files-v2#folders-for-organization), like so 
<img width="280" alt="image" src="https://user-images.githubusercontent.com/6067491/235431853-6471ef08-6b3e-4c83-8f79-fba383358abf.png">

This however results in the _posts/:slug_ subpath not being found

```zsh
npx remix routes

<Routes>
  <Route file="root.tsx">
    <Route path="healthcheck" file="routes/healthcheck.tsx" />
    <Route index file="routes/_index.tsx" />
    <Route path="logout" file="routes/logout.tsx" />
    <Route path="login" file="routes/login.tsx" />
    <Route path="notes" file="routes/notes.tsx">
      <Route path=":noteId" file="routes/notes.$noteId.tsx" />
      <Route index file="routes/notes._index.tsx" />
      <Route path="new" file="routes/notes.new.tsx" />
    </Route>
    <Route path="posts" file="routes/posts/route.tsx" />
    <Route path="join" file="routes/join.tsx" />
  </Route>
</Routes>
```

**Reply by kiliman** [reply]
Are you sure you're on v1.15.0 and have `v2_routeConvention: true`?
```
❯ tree app/routes                  
app/routes
└── posts
    ├── $slug
    │   └── route.tsx
    └── route.tsx
```
```ts
❯ npx remix routes                 
<Routes>
  <Route file="root.tsx">
    <Route path="posts" file="routes/posts/route.tsx">
      <Route path=":slug" file="routes/posts/$slug/route.tsx" />
    </Route>
  </Route>
</Routes>
```

**Reply by nlindroos** [reply]
Just created `npx create-remix@latest --template remix-run/indie-stack` as described in the [short tutorial](https://remix.run/docs/en/1.15.0/tutorials/blog), resulting in

```ts
// remix.config.js

/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
  cacheDirectory: "./node_modules/.cache/remix",
  future: {
    v2_errorBoundary: true,
    v2_meta: true,
    v2_normalizeFormMethod: true,
    v2_routeConvention: true,
  },
  ignoredRouteFiles: ["**/.*", "**/*.css", "**/*.test.{js,jsx,ts,tsx}"],
};

```


```
// package.json

  "dependencies": {
    "@remix-run/css-bundle": "^1.15.0",
    "@remix-run/node": "^1.15.0",
    "@remix-run/react": "^1.15.0",
    "@remix-run/serve": "^1.15.0",
    ...
  },
  "devDependencies": {
    "@remix-run/dev": "^1.15.0",
    "@remix-run/eslint-config": "^1.15.0",
    ...
```

**Reply by starkajs** [reply]
I'm on v.15.0 and have started a brand new project, no changes, using the indie-stack remix and for the life of me cannot get the folders to work in the manner you have them in the stackblitz example you have (https://stackblitz.com/edit/node-o2w958?file=app%2Froutes%2Fposts%2F%24slug%2Froute.tsx)

![image](https://github.com/remix-run/remix/assets/9750437/581bd710-2cf1-40c2-9e0f-d8d57d876d82)

Is the folder structure no longer supported? It's not really clear (to me) from the documentation as to how to structure the folders in the v.15.0. Can you point to a working example on the site somewhere as I can't see anything.

Thanks
Andrew

**Reply by kiliman** [reply]
Yes, apparently, Remix v2 routing no longer supports folders for organization; Only for colocation. The entire route name must be dot-separated.

```
routes/posts.$slug/route.tsx
routes/posts._index/route.tsx
```

You can use `remix-flat-routes` if you want to use folders for organization. This is called hybrid routes.

**Reply by yira-gallardo** [reply]
I had this same problem and struggled a lot to find answers.
The **_._index_** name made it work, but not a fan of that suffix.

**Reply by rex-empowerdigital** [reply]
Thanks @kiliman. Your suggestion works for me.

**Comment by kellyrmilligan** [comment]
had the same issue following the tutorial and ended up here. the tutorial should either have the command not enable the feature flags for v2 or be updated to use the new conventions most likely.

**Reply by wearypossum4770** [reply]
I was having an error in my system. Seems like when I try to use dynamic routing my terminal automagically thinks it is a  shell variable and removes it. as you can see I attempted to move the `posts..tsx` file to the `posts.$slug.tsx`  but my system recognizes them as the same. Maybe this routing idea wasn't the best. Could we get some type of modification with a different character maybe "@" or ":" .
![image](https://github.com/remix-run/remix/assets/50812322/70705b1b-8c4d-49c4-9ec5-c6bca41d1155)

**Reply by kiliman** [reply]
You need to escape the `$` in the shell by using `\$`. In my package `remix-flat-routes` I have an option to change the param prefix.

https://github.com/kiliman/remix-flat-routes#custom-param-prefix
